pkg/hypervctl/system_settings.go

```diff
@@ -4,204 +4,366 @@
+		params = append(params, fmt.Sprintf("-SnapshotFileLocation '%s'", opts.SnapshotFileLocation))
+	}
+	if opts.Notes != "" {
+		params = append(params, fmt.Sprintf("-Notes '%s'", opts.Notes))
 	}
-	defer service.Close()
 
-	if err = populateDefaults(resourceType, settings); err != nil {
-		return err
+	// Numeric parameters
+	if opts.ProcessorCount > 0 {
+		params = append(params, fmt.Sprintf("-ProcessorCount %d", opts.ProcessorCount))
+	}
+	if opts.AutomaticStartDelay > 0 {
+		params = append(params, fmt.Sprintf("-AutomaticStartDelay %d", opts.AutomaticStartDelay))
+	}
+	if opts.AutomaticCriticalErrorActionTimeout > 0 {
+		params = append(params, fmt.Sprintf("-AutomaticCriticalErrorActionTimeout %d", opts.AutomaticCriticalErrorActionTimeout))
 	}
 
-	if cb != nil {
-		cb()
+	// Memory parameters (in bytes)
+	if opts.MemoryStartupBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryStartupBytes %d", opts.MemoryStartupBytes))
+	}
+	if opts.MemoryMinimumBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryMinimumBytes %d", opts.MemoryMinimumBytes))
+	}
+	if opts.MemoryMaximumBytes > 0 {
+		params = append(params, fmt.Sprintf("-MemoryMaximumBytes %d", opts.MemoryMaximumBytes))
+	}
+	if opts.HighMemoryMappedIoSpace > 0 {
+		params = append(params, fmt.Sprintf("-HighMemoryMappedIoSpace %d", opts.HighMemoryMappedIoSpace))
+	}
+	if opts.LowMemoryMappedIoSpace > 0 {
+		params = append(params, fmt.Sprintf("-LowMemoryMappedIoSpace %d", opts.LowMemoryMappedIoSpace))
 	}
 
-	resourceStr, err := createResourceSettingGeneric(settings, resourceType)
-	if err != nil {
-		return err
+	// Boolean parameters
+	if opts.DynamicMemory {
+		params = append(params, "-DynamicMemory $true")
+	}
+	if opts.StaticMemory {
+		params = append(params, "-StaticMemory $true")
+	}
+	if opts.AutomaticCheckpointsEnabled {
+		params = append(params, "-AutomaticCheckpointsEnabled $true")
+	}
+	if opts.AllowUnverifiedPaths {
+		params = append(params, "-AllowUnverifiedPaths $true")
+	}
+	if opts.LockOnDisconnect {
+		params = append(params, "-LockOnDisconnect $true")
+	}
+	if opts.Passthru {
+		params = append(params, "-Passthru")
 	}
 
-	path, err := addResource(service, s.Path(), resourceStr)
-	if err != nil {
-		return err
+	// Enum parameters
+	if opts.AutomaticStartAction != "" {
+		params = append(params, fmt.Sprintf("-AutomaticStartAction %s", opts.AutomaticStartAction))
+	}
+	if opts.AutomaticStopAction != "" {
+		params = append(params, fmt.Sprintf("-AutomaticStopAction %s", opts.AutomaticStopAction))
+	}
+	if opts.AutomaticCriticalErrorAction != "" {
+		params = append(params, fmt.Sprintf("-AutomaticCriticalErrorAction %s", opts.AutomaticCriticalErrorAction))
+	}
+	if opts.CheckpointType != "" {
+		params = append(params, fmt.Sprintf("-CheckpointType %s", opts.CheckpointType))
+	}
+	if opts.GuestControlledCacheTypes != "" {
+		params = append(params, fmt.Sprintf("-GuestControlledCacheTypes %s", opts.GuestControlledCacheTypes))
```

Comment:
this should be a boolean? https://learn.microsoft.com/en-us/powershell/module/hyper-v/set-vm?view=windowsserver2025-ps#-guestcontrolledcachetypes